### PR TITLE
getting correct xds_root_cert info for config with the same logic of FindRootCAForXDS in Istio

### PR DESCRIPTION
**Pull request by zhlsunshine**
getting correct `xds_root_cert` info for config with the same logic of `FindRootCAForXDS` in Istio

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -353,6 +373,45 @@ pub fn empty_to_none<A: AsRef<str>>(inp: Option<A>) -> Option<A> {
     inp
 }
 
+pub fn find_rootca_for_xds(citadel_ca_cert_path: String) -> Result<RootCert, Error> {
```
Comment (howardjohn) [inline src/config.rs:376]: The existing istio implementation is legacy - we have to handle all sorts of old junk. I don't think we need most of this at all today. Probably just default to `./var/run/secrets/istio/root-cert.pem` and allow a single env to override.

BTW: ca_root_cert should have similar logic (but separate config)

Comment (zhlsunshine) [reply src/config.rs:376]: Hi @howardjohn, okay, There should be 2 envs for xds and ca root cert, such as `XDS_CERT_PROVIDER` for xds root cert and  `CA_CERT_PROVIDER` for ca root cert. And both of them have `./var/run/secrets/istio/root-cert.pem` by default.

**Comment by zhlsunshine** [comment]
Hi @howardjohn, re-submitted, and the change includes both xds and ca root cert.

File: src/config.rs [hunk 2]
```diff
@@ -48,6 +48,11 @@ const DEFAULT_CLUSTER_ID: &str = "Kubernetes";
 
 const ISTIO_META_PREFIX: &str = "ISTIO_META_";
 
+/// Fetch the XDS/CA root cert file path based on below constants
+const XDS_ROOT_CERT_PROVIDER_ENV: &str = "XDS_CERT_PROVIDER";
```
Comment (howardjohn) [inline src/config.rs:52]: I would name it XDS_ROOT_CA and CA_ROOT_CA to match the istio-agent

File: src/config.rs [hunk 3]
```diff
@@ -195,6 +200,26 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
         Some(parse_default(CA_ADDRESS, default_istiod_address)?)
     });
 
+    let xds_root_cert_provider = parse_default(
+        XDS_ROOT_CERT_PROVIDER_ENV,
+        DEFAULT_ROOT_CERT_PROVIDER.to_string(),
+    )?;
+    let xds_root_cert = if Path::new(&xds_root_cert_provider).exists() {
```
Comment (howardjohn) [inline src/config.rs:207]: can we handle SYSTEM special case as well?

File: src/config.rs [hunk 4]
```diff
@@ -195,6 +200,26 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
...
+        RootCert::File(xds_root_cert_provider.parse().unwrap())
```
Comment (howardjohn) [inline src/config.rs:208]: unwrap here isn't safe, can we propagate the error?

Comment (howardjohn) [reply src/config.rs:208]: It was fine before since it was for a constant

**Review by howardjohn — APPROVED** [review]
LGTM, we can address the rest in followup

File: src/config.rs (hunk 3 above)
Comment (zhlsunshine) [reply src/config.rs:207]: Okay, will fix it later

File: src/config.rs (hunk 2 above)
Comment (zhlsunshine) [reply src/config.rs:52]: sure, it makes sense to match the isito-agent

File: src/config.rs (hunk 4 above)
Comment (zhlsunshine) [reply src/config.rs:208]: sure, make sure no panic here.

**Pull request merged**